pulsar-log4j2-appender/src/main/java/org/apache/pulsar/log4j2/appender/PulsarAppender.java

```diff
@@ -186,7 +185,7 @@ public void start() {
         try {
             manager.startup();
         } catch (Exception e) {
-            // fail to start the manager
+            LOGGER.error("Failed to start pulsar manager", e);
```

Comment:
should we throw a RuntimeException ?

if you configure the logger to write to Pulsar and we are not able to setup the Pulsar client we are going to lose data

Reply:
see here https://github.com/apache/pulsar-adapters/pull/17/files#diff-edfe43967b455a71721fe0569016ff77ad3118a25c3616a342500b2f837bd9b9L130

Reply:
I didn't change the external behavior of the previous solution. I simply removed the catch in the location which you linked to and moved the logging to the upper level (here) since it reduces the confusion about swallowing the exception and not logging it.

It would be a breaking change to throw the exception as a RuntimeException. For example, if Pulsar is unavailable the application wouldn't be able to start. This might not be the desired behavior. I think changing the behavior would require a new configuration option and that would be a separate PR.

Reply:
makes sense